chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc

```diff
@@ -285,12 +285,14 @@ void BraveRenderViewContextMenu::ExecuteCommand(int id, int event_flags) {
       break;
 #endif
 #if BUILDFLAG(ENABLE_TOR)
-    case IDC_CONTENT_CONTEXT_OPENLINKTOR:
-      TorProfileManager::SwitchToTorProfile(
-          GetProfile(),
-          base::BindRepeating(OnTorProfileCreated, params_.link_url,
-                              HasAlreadyOpenedTorWindow(GetProfile())));
-      break;
+    case IDC_CONTENT_CONTEXT_OPENLINKTOR: {
+      const bool has_tor_window = HasAlreadyOpenedTorWindow(GetProfile());
+      Browser* tor_browser =
+          TorProfileManager::SwitchToTorProfile(GetProfile());
+      if (tor_browser) {
+        OnTorProfileCreated(params_.link_url, has_tor_window, tor_browser);
```

Comment:
won't this create an unnecessary tab after you added `chrome::NewTab(browser);`?

Reply:
No

Reply:
> No

can you elaborate please?

Reply:
Because has_tor_window = false in case we're opening a new tor window, it means use_new_tab parameter == false in the OnTorProfileCreated and it leads to disposition == WindowOpenDisposition::CURRENT_TAB